icu4c/source/i18n/calendar.cpp

```diff
@@ -4247,14 +4247,11 @@ Calendar::internalSet(EDateFields field, int32_t value)
     internalSet(static_cast<UCalendarDateFields>(field), value);
 }
 
-int32_t Calendar::internalGetMonth(UErrorCode& status) const {
-    if (U_FAILURE(status)) {
-        return 0;
-    }
+int32_t Calendar::internalGetMonth(UErrorCode& /* status */ ) const {
     if (resolveFields(kMonthPrecedence) == UCAL_MONTH) {
-        return internalGet(UCAL_MONTH, status);
+        return internalGet(UCAL_MONTH);
     }
-    return internalGet(UCAL_ORDINAL_MONTH, status);
+    return internalGet(UCAL_ORDINAL_MONTH);
```

Comment:
Why does the corresponding line in the following overload not pass in the defaultValue?